### PR TITLE
Replace regex with fancy-regex

### PR DESCRIPTION
**Pull request by fhennig**
## Description

I want to use look-ahead in a unit regex, and the `regex` package does not support this. `fancy-regex` is built on top of the `regex` package and supports look-ahead.

This PR replaces `regex` with `fancy-regex`

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -89,6 +89,14 @@ pub enum Error {
     #[error("Invalid regex pattern for unit '{unit}': '{regex}'")]
     InvalidRegexPattern { unit: String, regex: String },
 
+    #[error("lala")]
```
Comment (maltesander) [inline src/error.rs:92]: Proper error message?

**Review by maltesander — APPROVED** [review]
LGTM.

**Comment by teozkr** [comment]
Which operator needs this, by the way?

**Comment by fhennig** [comment]
Maybe I went overboard, but I wanted to have a regex for a iso8601 duration, which is used in druid to specify durations

**Pull request merged**